impeller/renderer/backend/gles/texture_gles.cc

```diff
@@ -281,9 +284,9 @@ bool TextureGLES::OnSetContents(std::shared_ptr<const fml::Mapping> mapping,
       tex_data = data->data->GetMapping();
     }
 
-    {
-      TRACE_EVENT1("impeller", "TexImage2DUpload", "Bytes",
-                   std::to_string(data->data->GetSize()).c_str());
+    TRACE_EVENT1("impeller", "TexImage2DUpload", "Bytes",
+                 std::to_string(data->data->GetSize()).c_str());
+    if (region == IRect::MakeLTRB(0, 0, size.width, size.height)) {
```

Comment:
This is unnecessary. TexSubImage2D with the whole dimensions should be the same as a TexImage2D. Unnecessarily detracts from the readability IMO.

Reply:
We have to call TexImage2D at least once, but we can do that will nullptr data and make it conditional on contents_initialized.